homework05/test/examples/Main.hs

```diff
@@ -1,15 +1,63 @@
-module Main
-  ( main
-  )
-  where
-
--- doctest
-import qualified Test.DocTest as DocTest
+module Main 
+  ( main )
+where
 
+import ExprT
+import Calc
+import Parser
+import StackVM
+import Test.Hspec
 
 main :: IO ()
-main =
-  DocTest.doctest
-    [ "-isrc"
-    , "src/Calc.hs"
-    ]
+main = hspec $ do
+  
+  describe "eval" $
+    it "returns result of applying add and multiply operations" $
+      eval (ExprT.Mul (ExprT.Add (ExprT.Lit 2) (ExprT.Lit 3)) (ExprT.Lit (-4))) `shouldBe` (-20)
+
+  describe "evalStr" $ do
+    it "evaluates expression with parenthesis" $
```

Comment:
Added two nested contexts because when the expression is well-formed (with or without parentheses), the function return value of the expression and when the expression is not well-formed (with or without parentheses), return `Nothing`.